sympy/integrals/intpoly.py

```diff
@@ -1009,19 +1009,26 @@ def point_sort(poly, normal=None, clockwise=True):
     >>> point_sort([Point(0, 0), Point(1, 0), Point(1, 1)])
     [Point2D(1, 1), Point2D(1, 0), Point2D(0, 0)]
     """
-    n = len(poly)
+    flag = 0
```

Comment:
I would recommend returning a list only. At the outset, convert Polygon to vertices instead of poly and then no flag will be needed and the code will use vertices instead of poly.vertices.

```python
if isinstance(poly, Polygon):
    pts = poly.vertices
else:
    pts = poly
n = len(pts)
...
future reference to poly.vertices becomes pts and final lines at 1064 are left unchanged except for changing poly to pts
```

Reply:
@smichr I was using flag to differentiate when to return Polygon object and when to return a list.
Because when we pass a list of points, then it must return a list (as shown in line 1009 in integrals/intpoly.py) whereas when we set clockwise=True in polytope_integrate() then only point_sort() is called and must return a Polygon type object for further evaluation.